RBDM/Record.cpp

```diff
@@ -0,0 +1,83 @@
+#include"Record.h"
+
+Record::Record()
+{
+	id = ++ID;
+}
+
+void Record::Add(char type, void* value)
+{
+	switch (type)
+	{
+	case 'i':
+	{
+		Integer *cell = (Integer*)value;
+		record.push_back(cell);
+		break;
+	}
+	case 'd':
+	{
+		Double *cell = (Double*)value;
+		record.push_back(cell);
+		break;
+	}
+	case 's':
+	{
+		String *cell = (String*)value;
+		record.push_back(cell);
+		break;
+	}
+	default:
+		break;
+	}
+}
+
+void Record::Show()
+{
+	for (int i = 0; i < record.size(); i++)
+	{
+		if (typeid(*record[i]) == typeid(Integer))
+			std::cout << *(Integer*)record[i] << " ";
+		if (typeid(*record[i]) == typeid(Double))
+			std::cout << *(Double*)record[i] << " ";
+		if (typeid(*record[i]) == typeid(String))
+			std::cout << *(String*)record[i] << " ";
+	}
+	std::cout << std::endl;
+}
+
+void Record::IdReset(int newid)
+{
+	id = newid;
+}
+
+void Record::Clear()
+{
+	for (unsigned i = 0; i < record.size(); i++)
```

Comment:
It seems that this won't work!
Did you ever launch this code?
Missed type specifier